src/runtime/typemanager.cs

```diff
@@ -817,4 +963,183 @@ internal static void CopySlot(IntPtr from, IntPtr to, int offset)
+            {
+                return;
+            }
+            var self = RecoverFromCapsule(capsule);
+            self.ResetSlots();
+            Runtime.XDecref(capsule);
+
+            IntPtr tp_dict = Marshal.ReadIntPtr(type, TypeOffset.tp_dict);
+            if (Runtime.PyDict_DelItemString(tp_dict, HolderKeyName) != 0)
+            {
+                throw new PythonException();
+            }
+        }
+
+        private void ResetSlots()
+        {
+            if (_alredyReset)
+            {
+                return;
+            }
+            _alredyReset = true;
+            IntPtr tp_name = Marshal.ReadIntPtr(_type, TypeOffset.tp_name);
+            string typeName = Marshal.PtrToStringAnsi(tp_name);
+            foreach (var offset in _slots.Keys)
+            {
+                IntPtr ptr = GetDefaultSlot(offset);
+                //DebugUtil.Print($"Set slot<{TypeOffsetHelper.GetSlotNameByOffset(offset)}> to 0x{ptr.ToString("X")} at {typeName}<0x{_type}>");
+                Marshal.WriteIntPtr(_type, offset, ptr);
+            }
+
+            foreach (var action in _deallocators)
+            {
+                action();
+            }
+
+            foreach (var pair in _customRestors)
+            {
+                int offset = pair.Key;
+                var resetor = pair.Value;
+                resetor?.Invoke(_type, offset);
+            }
+
+            _customRestors.Clear();
+            _slots.Clear();
+            _keepalive.Clear();
+            _deallocators.Clear();
+
+            // Custom reset
+            IntPtr tp_base = Marshal.ReadIntPtr(_type, TypeOffset.tp_base);
+            Runtime.XDecref(tp_base);
+            Marshal.WriteIntPtr(_type, TypeOffset.tp_base, IntPtr.Zero);
+
+            IntPtr tp_bases = Marshal.ReadIntPtr(_type, TypeOffset.tp_bases);
+            Runtime.XDecref(tp_bases);
+            Marshal.WriteIntPtr(_type, TypeOffset.tp_bases, IntPtr.Zero);
+
+            // Because __len__ may have a "default" implementation, not registered to us
+            // always explicitly clobber it. See mp_lenth_slot class
+            Marshal.WriteIntPtr(_type, TypeOffset.mp_length, IntPtr.Zero);
+        }
+
+        private static void OnDestruct(IntPtr ob)
+        {
+            var self = RecoverFromCapsule(ob);
+            self._handle.Free();
+            self.ResetSlots();
+        }
+
+        private static SlotsHolder RecoverFromCapsule(IntPtr ob)
+        {
+            var ptr = Runtime.PyCapsule_GetPointer(ob, null);
+            PythonException.ThrowIfIsNull(ptr);
+            GCHandle handle = GCHandle.FromIntPtr(ptr);
+            return (SlotsHolder)handle.Target;
+        }
+
+        private static IntPtr GetDefaultSlot(int offset)
+        {
+            if (offset == TypeOffset.tp_clear
+                || offset == TypeOffset.tp_traverse)
+            {
+                return TypeManager.NativeCodePage + TypeManager.NativeCode.Active.Return0;
```

Comment:
Maybe I missed a conversation. What is the story for ARM in this code?

Reply:
I'm not sure if this is what you're looking for, but it seems that it was removed in #913